### DISCUSSION TITLE
How to combine with barryvdh/laravel-ide-helper model docs and PHPStan

### DISCUSSION THREAD

**Discussion by SolveSoul** [spatie/laravel-translatable#365] [Q&A] [answered]
Hi all,

I'm looking for an optimised solution to balance the usage of ide-helper, phpstan and laravel-translatable packages. Let's dive in:

Translations are stored in json in the database which results in an array being returned for the translated property instead of a plain string. 

When using the package [barryvdh/laravel-ide-helper](https://github.com/barryvdh/laravel-ide-helper) and generating model docs the translated columns are correctly marked as array. For example:

```
/*
 * @property array $name
 */
class MyModel extends BaseModel
{
...
    public $translatable = [
        'name',
    ];
...
}
```

However PHPStan identifies the usage of the translated string as incorrect because the "array" is accessed as string
`$name = $model->name;` results in error `Part $model->name (array) of encapsed string cannot be cast to string.`

You can however mark the property on your model as `@property array|string $name` which works and makes PHPStan happy. But every time I run the command to update the model docs (`php artisan ide-helper:models --reset`) I need to restore all those manually edited typings which is inconvenient.

Any ideas for a better solution?

**Comment by sergeynilov** [comment]
I have similar issue with phpstan ...

**Comment by SolveSoul** [comment, marked as answer]
For what it's worth, I fixed the issue using a custom Model Hook. 

You'll need to publish the ide-helper config file using:

```bash
php artisan vendor:publish --provider="Barryvdh\LaravelIdeHelper\IdeHelperServiceProvider" --tag=config
```

Then you'll need to create a model hook class and add its reference to the `ide-helper.php` config file under  the `model_hooks` section.

See this gist for the code:

https://gist.github.com/SolveSoul/f9db8b9370d96c1aa4273b2280cb9b89

**Comment by ricventu** [comment]
thanks @SolveSoul
